test/functional/sqlsrv/sqlsrv_batch_query.phpt

```diff
@@ -0,0 +1,179 @@
+        fatalError("Got wrong number of errors\n");
+    }
+    
+    // Make sure the SQLSTATE, code, and message are identical
+    $e = 0;
+    foreach ($actualError AS $error) {
+        if ($error[0] != $expectedError[$e][0] or $error[1] != $expectedError[$e][1] or $error[2] != $expectedError[$e][2]) {
+            fatalError("Wrong error message, expected ".$error[2].", got ".$expectedError[$e][2]."\n");
+        }
+        ++$e;
+    }
+}
+
+function checkColumnsAndRows($stmt, $cursor, $error)
+{
+    global $expectedCols, $expectedRows;
+    
+    $cols = sqlsrv_num_fields($stmt);
+    
+    if ($cols != $expectedCols) {
+        fatalError("Incorrect number of columns returned with ".$cursor." cursor. Expected ".$expectedCols." columns, got ".$cols." columns\n");
+    }
+
+    $rows = sqlsrv_num_rows($stmt);
+
+    if ($cursor == 'buffered') {
+        if ($rows != $expectedRows) {
+            fatalError("Incorrect number of columns returned with buffered cursor. Expected ".$expectedRows." rows, got ".$rows." rows\n");
+        }
+    } else {
+        if ($rows !== false) {
+            fatalError("Expected sqlsrv_num_rows to return false with ".$cursor." cursor, instead returned ".$rows." rows\n");
+        } else {
+            checkErrors($error);
+        }
+    }
+}
+
+sqlsrv_configure('WarningsReturnAsErrors', 1);
+
+$conn = AE\connect();
+
+// Create and populate a table of integer types
+$tableName = 'batch_query_test';
+$columns = array(new AE\ColumnMeta('int', 'c1_int'),
+                 new AE\ColumnMeta('tinyint', 'c2_tinyint'),
+                 new AE\ColumnMeta('smallint', 'c3_smallint'),
+                 new AE\ColumnMeta('bigint', 'c4_bigint'),
+                 new AE\ColumnMeta('bit', 'c5_bit'));
+                 
+$stmt = AE\createTable($conn, $tableName, $columns);
+if (!$stmt) {
+    fatalError("Failed to create table $tableName\n");
+}
+sqlsrv_free_stmt($stmt);
+
+$inputs = array(array('c1_int'=>86, 'c2_tinyint'=>0, 'c3_smallint'=>4534, 'c4_bigint'=>-1, 'c5_bit'=>0),
+                array('c1_int'=>-217483648, 'c2_tinyint'=>31, 'c3_smallint'=>-212, 'c4_bigint'=>546098342985694, 'c5_bit'=>1),
+                array('c1_int'=>0, 'c2_tinyint'=>127, 'c3_smallint'=>32767, 'c4_bigint'=>9223372000000000000, 'c5_bit'=>0),
+                array('c1_int'=>-432987563, 'c2_tinyint'=>255, 'c3_smallint'=>0, 'c4_bigint'=>5115115115115115115, 'c5_bit'=>0),
+                array('c1_int'=>7, 'c2_tinyint'=>1, 'c3_smallint'=>7, 'c4_bigint'=>7, 'c5_bit'=>1),
+               );
+               
+for ($i=0; $i < sizeof($inputs); ++$i) {
+    $stmt = AE\insertRow($conn, $tableName, $inputs[$i]);
+    sqlsrv_free_stmt($stmt);
+}
+
+$query = "SELECT c1_int from batch_query_test;
+          SELECT c2_tinyint from batch_query_test;
+          SELECT c3_smallint from batch_query_test;
+          SELECT c4_bigint from batch_query_test;
+          SELECT c5_bit from batch_query_test;";
+
```

Comment:
Did you mean to use `$tableName` in `$query`? Also, to be consistent please use upper caps for tsql keywords